torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -1,71 +1,121 @@
-from typing import Tuple, List, Optional
+from typing import List, Optional, TypeVar
 
 import torch
-from torchvision.transforms import (  # noqa: F401
-    functional as _F,
-    InterpolationMode,
-)
+from torchvision.prototype import features
+from torchvision.prototype.transforms import kernels as K
+from torchvision.transforms import functional as _F, InterpolationMode
 
-horizontal_flip_image = _F.hflip
+from ._utils import dispatch
 
+T = TypeVar("T", bound=features.Feature)
 
-def horizontal_flip_bounding_box(bounding_box: torch.Tensor, *, image_size: Tuple[int, int]) -> torch.Tensor:
-    bounding_box = bounding_box.clone()
-    bounding_box[..., (0, 2)] = image_size[1] - bounding_box[..., (2, 0)]
-    return bounding_box
 
+def _horizontal_flip_bounding_box(input: features.BoundingBox) -> torch.Tensor:
+    return K.horizontal_flip_bounding_box(input, format=input.format, image_size=input.image_size)
 
-_resize_image = _F.resize
 
+@dispatch(
+    {
+        features.Image: K.horizontal_flip_image,
+        features.BoundingBox: _horizontal_flip_bounding_box,
+    },
+    pil_kernel=_F.hflip,
+)
+def horizontal_flip(input: T) -> T:
+    """ADDME"""
+    ...
 
-def resize_image(
-    image: torch.Tensor,
-    size: List[int],
-    interpolation: InterpolationMode = InterpolationMode.BILINEAR,
-    max_size: Optional[int] = None,
-    antialias: Optional[bool] = None,
-) -> torch.Tensor:
-    new_height, new_width = size
-    num_channels, old_height, old_width = image.shape[-3:]
-    batch_shape = image.shape[:-3]
-    return _resize_image(
-        image.reshape((-1, num_channels, old_height, old_width)),
-        size=size,
-        interpolation=interpolation,
-        max_size=max_size,
-        antialias=antialias,
-    ).reshape(batch_shape + (num_channels, new_height, new_width))
-
-
-def resize_segmentation_mask(
-    segmentation_mask: torch.Tensor,
-    size: List[int],
-    interpolation: InterpolationMode = InterpolationMode.NEAREST,
-    max_size: Optional[int] = None,
-    antialias: Optional[bool] = None,
-) -> torch.Tensor:
-    return resize_image(
-        segmentation_mask, size=size, interpolation=interpolation, max_size=max_size, antialias=antialias
-    )
-
-
-# TODO: handle max_size
-def resize_bounding_box(
-    bounding_box: torch.Tensor,
-    *,
-    old_image_size: List[int],
-    new_image_size: List[int],
-) -> torch.Tensor:
-    old_height, old_width = old_image_size
-    new_height, new_width = new_image_size
-    ratios = torch.tensor((new_width / old_width, new_height / old_height))
-    return bounding_box.view(-1, 2, 2).mul(ratios).view(bounding_box.shape)
 
+def _resize_bounding_box(input: features.BoundingBox, *, size: List[int]) -> features.BoundingBox:
```

Comment:
It is still ambiguous signatures for decorated function and implementations. As I understand dispatch decorates empty function and does completely another thing. Unfortunately, I do not understand the point of providing `resize` complete signature which can be unrelated to the implementations signatures. I'd instead code it with *args, **kwargs and in the docs would detail that *args, **kwargs are specific to the implementations:
```python
@dispatch(
    {
        features.Image: K.resize_image,
        features.SegmentationMask: K.resize_segmentation_mask,
        features.BoundingBox: _resize_bounding_box,
    },
    pil_kernel=_F.resize,
)
def resize(input: T, *args: Any, **kwargs: Any) -> T:
    """ADDME"""
    ...
```